pkg/api/metrics.go

```diff
@@ -71,143 +38,19 @@ func InitMetrics() *Metrics {
-				Help:      "Duration of sample batch send calls to the remote storage.",
-				Buckets:   prometheus.DefBuckets,
-			},
-		),
 		LastRequestUnixNano: time.Now().UnixNano(),
-		QueryRemoteReadBatchDuration: prometheus.NewHistogram(
-			prometheus.HistogramOpts{
-				Namespace: util.PromNamespace,
-				Subsystem: "metrics",
-				Name:      "query_remote_read_batch_duration_seconds",
-				Help:      "Duration of query batch read calls to the remote storage.",
-				Buckets:   []float64{0.05, 0.1, 0.25, 0.5, 1, 2.5, 5, 10, 30, 50, 100, 250, 500, 1000, 2500},
-			},
-		),
-		QueryDuration: prometheus.NewHistogram(
-			prometheus.HistogramOpts{
-				Namespace: util.PromNamespace,
-				Subsystem: "metrics",
-				Name:      "query_duration_seconds",
-				Help:      "Duration of query batch read calls to the PromQL engine.",
-				Buckets:   []float64{0.05, 0.1, 0.25, 0.5, 1, 2.5, 5, 10, 30, 50, 100, 250, 500, 1000, 2500},
-			},
-		),
-		ExemplarQueryDuration: prometheus.NewHistogram(
-			prometheus.HistogramOpts{
-				Namespace: util.PromNamespace,
-				Name:      "exemplar_query_duration_seconds",
-				Help:      "Duration of exemplar query read calls to the database.",
-				Buckets:   prometheus.DefBuckets,
-			},
-		),
-		FailedQueries: prometheus.NewCounter(
-			prometheus.CounterOpts{
-				Namespace: util.PromNamespace,
-				Subsystem: "metrics",
-				Name:      "queries_failed_total",
-				Help:      "Total number of queries which failed on send to remote storage.",
-			},
-		),
-		InvalidReadReqs: prometheus.NewCounter(
-			prometheus.CounterOpts{
-				Namespace: util.PromNamespace,
-				Name:      "invalid_read_requests",
-				Help:      "Total number of remote read requests with invalid metadata.",
-			},
-		),
-		InvalidWriteReqs: prometheus.NewCounter(
+		InvalidReqs: prometheus.NewCounterVec(
 			prometheus.CounterOpts{
 				Namespace: util.PromNamespace,
-				Name:      "invalid_write_requests",
-				Help:      "Total number of remote write requests with invalid metadata.",
-			},
-		),
-		InvalidQueryReqs: prometheus.NewCounter(
-			prometheus.CounterOpts{
-				Namespace: util.PromNamespace,
-				Name:      "invalid_query_requests",
-				Help:      "Total number of invalid query requests with invalid metadata.",
-			},
-		),
-		ReceivedQueries: prometheus.NewCounter(
-			prometheus.CounterOpts{
-				Namespace: util.PromNamespace,
-				Name:      "received_queries_total",
-				Help:      "Total number of received queries.",
-			},
-		),
-		ExecutedQueries: prometheus.NewCounter(
-			prometheus.CounterOpts{
-				Namespace: util.PromNamespace,
-				Name:      "executed_queries_total",
-				Help:      "Total number of successfully executed queries.",
-			},
-		),
-		TimedOutQueries: prometheus.NewCounter(
-			prometheus.CounterOpts{
-				Namespace: util.PromNamespace,
-				Name:      "queries_timed_out_total",
-				Help:      "Total number of timed out queries.",
-			},
+				Subsystem: "api",
+				Name:      "invalid_requests",
```

Comment:
I don't see this metric in the design doc

Reply:
It was earlier `Promscale_invalid_read_requests`, which is also in the design doc. Now, these have been categorized under `api` subsystem.